kokoro-tts

```diff
@@ -415,6 +417,130 @@ def extract_chapters_from_epub(epub_file, debug=False):
     
     return chapters
 
+
+
+class PdfParser:
+    def __init__(self, pdf_path):
+        self.pdf_path = pdf_path
+        self.chapters = []
+        
+
+
+    def get_chapters(self):
+        """
+        Get the chapters from the PDF file.
+        """
+        if self.get_chapter_from_toc():
+            return self.chapters
+        else:
+            self.chapters = self.get_chapters_from_txt()
+            return self.chapters
+
+
+    def get_chapter_from_toc(self):
+        """
+        Get the chapter from the table of contents.
+        Returns True if chapters were found, False otherwise.
+        """
+        doc = fitz.open(self.pdf_path)
+        toc = doc.get_toc()
+        if len(toc) == 0:
+            doc.close()
```

Comment:
use try..finally block to close pdf file as some error may lead to memory leeks or too many open file handles